### ISSUE TITLE
Should we use `std::shared_lock<>` to own the `db_mmap_lock_` in the View?

### ISSUE THREAD

**Issue by mapleFU** [yuyuaqwq/yudb#1]
Currently, `db_mmap_lock_` is allocated in `DBImpl::View`, and released in `TxManager::RollBack()`, the later one is called by `~ViewTx`. Should we use  `std::shared_lock<>` to better owning this?

**Comment by yuyuaqwq** [comment]
Yes, that's a great suggestion. Thank you

**Comment by mapleFU** [comment]
@yuyuaqwq Would Would [this](https://github.com/yuyuaqwq/yudb/commit/049f9df9319362d2c0b61510c1dffeb532ef266e) causing it Release twice? Since `TxManager::RollBack()` might call this explicitly?

**Comment by yuyuaqwq** [comment]
@mapleFU  No, because the Rollback of the View is only called in ~View, the user is not allowed to explicitly call

**Comment by mapleFU** [comment]
Acquire:

```
ViewTx DBImpl::View() {
     db_mmap_lock_.lock_shared();
     return tx_manager_->View();
 }
```

And:

```
TxManager::View() <-- acquire from ctor
```

Release:

```c++
void TxManager::RollBack(TxId view_txid) {
    db_->db_file_mmap_lock().unlock_shared();
```

And dtor.

Actually the code works but might allocate / release the lock twice?

**Comment by yuyuaqwq** [comment]
@mapleFU I'm sorry, I misunderstood your meaning.
Yes, this is my question. I forgot to remove it. Thank you for pointing out this issue

**Comment by mapleFU** [comment]
No problem, thanks for your quick response
